### ISSUE TITLE
http://umbrel.local/settings don't save the config

### ISSUE THREAD

**Issue by TZocker** [getumbrel/umbrel#1893]
I have the problem, that the temperature and language isn't saved permanently.

Only for the session. 

![settings](https://github.com/user-attachments/assets/2aef0921-73ae-43f1-b323-8c204233324a)

**Comment by JoseMoranUrena523** [comment]
So I’m pretty sure your issue is related to [this file](https://github.com/getumbrel/umbrel/blob/master/packages/ui/src/routes/live-usage.tsx). I might do a pull request later this weekend or next week and see if it can be included in something like umbrel.yaml.

**Comment by JoseMoranUrena523** [comment]
The CPU setting looks like its being saved correctlye, but I see the issue with the language setting noted in #1899. Make sure your browser allows local storage and session storage to be saved. If the pull request gets merged into the master branch and you keep getting issues, let me know and I'll continue looking or hopefully a developer at Umbrel can see this.

**Comment by TZocker** [comment]
> The CPU setting looks like its being saved correctlye, but I see the issue with the language setting noted in #1899. Make sure your browser allows local storage and session storage to be saved. If the pull request gets merged into the master branch and you keep getting issues, let me know and I'll continue looking or hopefully a developer at Umbrel can see this.

Yes this is correct but, i think the temperature setting is missing (F° vs. C°).

THX

**Comment by JoseMoranUrena523** [comment]
I'll take a look at the code for temperature in maybe an hour.

**Comment by JoseMoranUrena523** [comment]
So the specific code that handles that is this:
```ts
export function useTempUnit(optionalUnit?: TempUnit): [unit: TempUnit, setTemp: (unit: TempUnit) => void] {
	// Get default unit from the user's language
	const [langCode] = useLanguage()
	const langUnit = langCodesWithFahrenheitTemp.includes(langCode) ? 'f' : 'c'

	const defaultUnit = optionalUnit ?? langUnit

	// Not setting a default in `useLocalStorage2` because it would set the local storage value, which we don't want.
	// We want the unit to depend on the user language until the user specifically toggles to the a different temp unit.
	// Once they set the unit they want, we don't wanna change it even if they change the language.
	const [unit, setUnit] = useLocalStorage2<TempUnit>('temp-unit')

	return [unit ?? defaultUnit, setUnit]
}
```
The way that it's handled is that it gets the temperature setting (C or F) based on your language or based on a setting saved in window.LocalStorage (represented by the useLocalStorage2 function). Me personally, I don't see any issue with the code, but it might be best for an Umbrel developer to check it out.

**Comment by TZocker** [comment]
Thx for the fast code review.

But we have to recheck the behavior if https://github.com/getumbrel/umbrel/pull/1899 is fixed.

Until that i have everytime a language change ...

**Comment by JoseMoranUrena523** [comment]
And we can’t set up a development environment sadly. Let’s hope that fixes one part of the problem (although Umbrel staff is taking a while).
